test/unit/provisioning_template_test.rb

```diff
@@ -141,21 +141,49 @@ def setup
       end
     end
 
-    context 'map_oses' do
+    context 'map_metadata' do
       test 'returns OSes that are in the db' do
         metadata = { 'oses' => ['centos 5.3', 'Fedora 19'] }
         assert_equal [Operatingsystem.find_by_title('centos 5.3')],
-                     ProvisioningTemplate.map_oses(metadata)
+                     ProvisioningTemplate.map_metadata(metadata, 'oses')
+      end
+
+      test 'returns Orgs that are in the db' do
+        metadata = { 'organizations' => ['Organization 1', 'Organization One'] }
+        assert_equal [Organization.find_by_name('Organization 1')],
+                     ProvisioningTemplate.map_metadata(metadata, 'organizations')
+        # test organisations as well as organizations metadata
+        metadata = { 'organisations' => ['Organization 1', 'Organization One'] }
+        assert_equal [Organization.find_by_name('Organization 1')],
+                     ProvisioningTemplate.map_metadata(metadata, 'organisations')
+      end
+
+      test 'returns Locations that are in the db' do
+        metadata = { 'locations' => ['Location 1', 'Location One'] }
+        assert_equal [Location.find_by_name('Location 1')],
+                     ProvisioningTemplate.map_metadata(metadata, 'locations')
       end
 
       test 'returns an empty array for no matched OSes' do
         metadata = { 'oses' => ['Fedora 19'] }
-        assert_equal [], ProvisioningTemplate.map_oses(metadata)
+        assert_equal [], ProvisioningTemplate.map_metadata(metadata, 'oses')
```

Comment:
can we DRY these tests in a new TemplateTest class? Rather than copying these changes to both files...